clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -136,6 +136,7 @@ public AbstractCoordinator(LogContext logContext,
                                Time time,
                                long retryBackoffMs,
                                boolean leaveGroupOnClose) {
+        assert groupId != null : "Expected a non-null group id for coordinator construction";
```

Comment:
nit: can we use `Objects.requireNonNull` instead?

Reply:
Made this change too.